### PR TITLE
[WIP]: Generalized Transfer Learning across MolNet

### PR DESCRIPTION
**Pull request by vsomnath**
Couple of scripts that allow model agnostic transfer learning on MolNet dataset (and extendable to others with the same style as MolNet)

`compute_loss` was currently added to KerasModel, because I need it for the EarlyStopping check. Will move the definition into a Callback style setup later.

### CODE REVIEW

**Comment by vsomnath** [comment]
Starting a discussion on this. @rbharath @peastman 

1. The `compute_loss` function needs to be changed and moved into an `EarlyStoppingCallback`. What are the metrics that `EarlyStoppingCallback` should support, apart from the loss?

2. The S3 bucket (or an equivalent hosting service) can only host only certain hyperparameter configurations for some model classes. If we have to download a pretrained model, the user should know what model classes are available and available hyperparameter settings (can resort to a default if nothing is provided). Do you store it with something like:

```
configs = {'ChemCeption': ['filters_16_blocklayers_3_imgspec_std', ...], 'GraphConv': ...}
```
or
```
configs = {'ChemCeption': [{'filters: 16, 'blocklayers': 3, 'imgspec':  'std'}]}
```

**Comment by peastman** [comment]
Would it make sense either to add early stopping as a feature of ValidationCallback, or alternatively to have a separate EarlyStoppingCallback that makes use of a ValidationCallback?  Computing metrics for a validation set is precisely what ValidationCallback already does.

**Comment by vsomnath** [comment]
I was thinking of the latter based on how Keras has it right now.

**Comment by vsomnath** [comment]
Usually early stopping is based on loss, and ValidationCallBack should either be extended to handle loss compute or make EarlyStoppingCallback a different class.

**Comment by peastman** [comment]
In my experience, early stopping is more often based on some other metric.  Loss functions often aren't appropriate for evaluating a model.  For example, they may include regularization terms that you hope will prevent it from overfitting to the training set, but that you wouldn't include when evaluating it on the validation set.

**Comment by rbharath** [comment]
For 2., the option

```
configs = {'ChemCeption': [{'filters: 16, 'blocklayers': 3, 'imgspec':  'std'}]}
```

Looks a little cleaner to me. Great to see this PR moving forward :)
